### ISSUE TITLE
DSM 7.2 the 2FA is now mandatory which looks like it is yet not supported by this version

### ISSUE THREAD

**Issue by tzago** [SynologyOpenSource/synology-csi#62]
Looks like 2FA is now mandatory and my csi user with admin group right fails to connect to DSM cause it passes only the first phase of the 2FA authentication as seen in the logs....

Trying to make my Synology CSI ISCSI work but not getting it through.

I0906 20:28:00.271711       1 event.go:282] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"default", Name:"busybox-pvc-tshoot-iscsi-03", UID:"6aa6fcf4-50b0-43ab-bd6c-xxxxxxxx", APIVersion:"v1", ResourceVersion:"637548", FieldPath:""}): type: 'Warning' reason: 'ProvisioningFailed' failed to provision volume with StorageClass "synostorage": rpc error: code = Internal desc = Couldn't find any host available to create Volume
I0906 20:28:00.272002       1 event.go:282] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"default", Name:"busybox-pvc-tshoot-iscsi-01", UID:"8a9e2772-49f5-402a-a7ad-b32034xxxxxxx", APIVersion:"v1", ResourceVersion:"637588", FieldPath:""}): type: 'Warning' reason: 'ProvisioningFailed' failed to provision volume with StorageClass "synology-iscsi-storage": rpc error: code = Internal desc = Couldn't find any host available to create Volume

#below the synology log showing my worker node trying to connect to the DSM. Only first auth passed via password.
`09/06/2023
13:59:27
Info
synology02
synology-k3s-csi
Connection
User [synology-k3s-csi] from [192.168.0.39] has successfully passed the first authentication of 2FA via [password]
09/06/2023
13:59:26
Info
synology02
synology-k3s-csi
Connection
User [synology-k3s-csi] from [192.168.0.39] has successfully passed the first authentication of 2FA via [password]
09/06/2023
13:59:26
Info
synology02
synology-k3s-csi
Connection
User [synology-k3s-csi] from [192.168.0.39] has successfully passed the first authentication of 2FA via [password]
09/06/2023
13:59:25
Info
synology02
synology-k3s-csi
Connection
User [synology-k3s-csi] from [192.168.0.39] has successfully passed the first authentication of 2FA via [password]
09/06/2023
13:58:40
Info
synology02
SYSTEM
System
System successfully stopped [SSH service].`

**Comment by laghoule** [comment]
@tzago it's possible to activate 2FA only on some users:
![2023-09-19_19-51](https://github.com/SynologyOpenSource/synology-csi/assets/1598055/10b1f03a-bd9c-4255-ab26-7ab19d7731c8)

**Comment by tzago** [comment]
Thanks a log @laghoule that worked for me.
Hey maybe what is missing in the docs is a comprehensive explanation on how to best create a user for this synology csi.
What is not clear from current documentation is the right settings for the user to be able to creat iscsi luns which clearly requires admin rights. 
Also what is needed to mitigate security risk while creating this user as the secret will be exposed in plain sight with the user synology user credentials.

**Comment by theobjectivedad** [comment]
I totally agree on this. I spent more time than I'm proud of getting the service account configured and hardened (as much as it is possible). For folks struggling to test you can run this directly from the csi-plugin container to test authentication:


```shell
# For TLS
curl -k https://SYNOLOGY_IP:5001/webapi/auth.cgi \
    -d "api=SYNO.API.Auth" \
    -d "method=login" \
    -d "version=3" \
    -d "account=USERNAME" \
    -d "passwd=PASSWORD" \
    -d "format=sid"

# Non-TLS
curl  http://SYNOLOGY_IP:5000/webapi/auth.cgi \
    -d "api=SYNO.API.Auth" \
    -d "method=login" \
    -d "version=3" \
    -d "account=USERNAME" \
    -d "passwd=PASSWORD" \
    -d "format=sid"
```

Note that I needed to use 5001 for TLS and couldn't get 443 working properly.
